### PR TITLE
Implement the 'close' command to close channels in Charmhub (CRAFT-493).

### LINKED ISSUE #232: Implement `charmcraft close`
There needs to be a way to close channels, bringing over syntax from snapcraft, the following syntax should be possible:

    charmcraft close <name> <channel>...

### PR DESCRIPTION
**Pull request by facundobatista**
Fixes #232.

### CODE REVIEW

File: charmcraft/commands/store/__init__.py [hunk 1]
```diff
@@ -560,6 +560,43 @@ def run(self, parsed_args):
...
+
+        The channel is made up of `track/risk/branch` with both the track and
+        the branch as optional items, so formally:
+
+          [track/]risk[/branch]
+
+        Channel risk must be one of stable, candidate, beta or edge. The
+        track defaults to `latest` and branch has no default.
+
+        Closing a channel will take you through login if needed.
```
Comment (sergiusens) [inline charmcraft/commands/store/__init__.py:580]: we will need to update this if we go down the agreed path of saying "Not logged in" if the user has never logged in before.

File: charmcraft/commands/store/__init__.py [hunk 2]
```diff
@@ -560,6 +560,43 @@ def run(self, parsed_args):
...
+
+    def fill_parser(self, parser):
+        """Add own parameters to the general parser."""
+        parser.add_argument("name", help="The name of charm or bundle")
+        parser.add_argument("channel", help="The channel to close")
+
+    def run(self, parsed_args):
+        """Run the command."""
+        store = Store(self.config.charmhub)
+        revision = None  # revision in None will actually close the channel
```
Comment (sergiusens) [inline charmcraft/commands/store/__init__.py:593]:
```suggestion
        revision = None  # revision None will actually close the channel
```

File: charmcraft/commands/store/__init__.py [hunk 3]
```diff
@@ -560,6 +560,43 @@ def run(self, parsed_args):
...
+        channels = [parsed_args.channel]  # the API accepts multiple channels, we have only one
+        resources = []  # no really used when closing channels
```
Comment (sergiusens) [inline charmcraft/commands/store/__init__.py:595]:
```suggestion
        resources = []  # not really used when closing channels
```

**Review by sergiusens — APPROVED** [review]
Looks good, just a few literal nits

File: charmcraft/commands/store/__init__.py (hunk 1 above)
Comment (facundobatista) [reply charmcraft/commands/store/__init__.py:580]: IIUC we should ask automatically for login in all operations except the auth-related ones (like `whoami`).

File: charmcraft/commands/store/__init__.py (hunk 3 above)
Comment (facundobatista) [reply charmcraft/commands/store/__init__.py:595]: yes, thanks!

**Review by cmatsuoka — APPROVED** [review]
LGTM.

**Pull request merged**